ratings/api/app.py

```diff
@@ -0,0 +1,100 @@
+from flask import Flask, g, request
+from flask_expects_json import expects_json
+from datetime import datetime, timezone
+import logging
+from logging import Formatter, FileHandler
+import psycopg2
+import os
+from utils import execute_query
+
+logger = logging.getLogger('babylon-ratings')
+
+CREATE_RATINGS_TABLE = """CREATE TABLE IF NOT EXISTS ratings (
+                    provision_uuid varchar(64) NOT NULL, email varchar NOT NULL, 
+                        catalog_item_id int NOT NULL, rating int, comment TEXT, 
+                        created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+                        updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+                        PRIMARY KEY(provision_uuid, email),
+                        FOREIGN KEY(provision_uuid) REFERENCES provisions(uuid) ON DELETE CASCADE,
+                        FOREIGN KEY(email) REFERENCES students(email) ON DELETE CASCADE,
+                        FOREIGN KEY(catalog_item_id) REFERENCES catalog_items(id) ON DELETE CASCADE
+                    );"""
```

Comment:
Hrm... It seems a bit odd to me having the application manage the database schema. I would expect this sort of thing to be elsewhere? As a database administrator I wouldn't want applications creating tables or even having permissions to createt tables.

Reply:
IMHO there are benefits in having this logic here, for example, if a new version of the application needs a table with the current data then you need to create this table at the specific moment of deploying the new version as you will need the data from that instant.
Also having encapsulated this logic in the application gives the developer a better understanding of the model.
What I don't like is sharing this DB between two services as it contains foreign keys related to other tables, but I guess is a compromise we need to take (as having a new database will mean keeping in sync those tables. And that is a problem that worth to be avoided if possible)

Do you see this as a vulnerability risk?